From 9a5e291b4f73bf1e8d3405df96b7b57e0f0442c3 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Wed, 19 Apr 2023 14:18:44 +0200
Subject: [PATCH 0206/1141] ASoC: dt-bindings: add stm32mp25 support for sai

Add STM32MP25 support for STM32 SAI peripheral,
through "st,stm32mp25-sai" compatible.

Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Change-Id: I9b51bbde99720e7e2d8ec8efabf9a5bfbbb81052
---
 .../bindings/sound/st,stm32-sai.yaml          | 29 ++++++++++++++++++-
 1 file changed, 28 insertions(+), 1 deletion(-)

--- a/Documentation/devicetree/bindings/sound/st,stm32-sai.yaml
+++ b/Documentation/devicetree/bindings/sound/st,stm32-sai.yaml
@@ -20,6 +20,7 @@ properties:
     enum:
       - st,stm32f4-sai
       - st,stm32h7-sai
+      - st,stm32mp25-sai
 
   reg:
     items:
@@ -43,9 +44,11 @@ properties:
     const: 1
 
   clocks:
+    minItems: 1
     maxItems: 3
 
   clock-names:
+    minItems: 1
     maxItems: 3
 
 required:
@@ -141,6 +144,7 @@ allOf:
         compatible:
           contains:
             const: st,stm32f4-sai
+
     then:
       properties:
         clocks:
@@ -152,7 +156,14 @@ allOf:
           items:
             - const: x8k
             - const: x11k
-    else:
+
+  - if:
+      properties:
+        compatible:
+          contains:
+            const: st,stm32mph7-sai
+
+    then:
       properties:
         clocks:
           items:
@@ -166,6 +177,22 @@ allOf:
             - const: x8k
             - const: x11k
 
+  - if:
+      properties:
+        compatible:
+          contains:
+            const: st,stm32mp25-sai
+
+    then:
+      properties:
+        clocks:
+          items:
+            - description: pclk feeds the peripheral bus interface.
+
+        clock-names:
+          items:
+            - const: pclk
+
 additionalProperties: false
 
 examples:
